### PR TITLE
Add Immutable.js support

### PR DESCRIPTION
**Pull request by evgeny-matveev**
Hello, I use your component and everything works fine, but I need https://facebook.github.io/immutable-js support in my application. In this PR I've tried to add one, what do you think?

### CODE REVIEW

**Comment by mathieudutour** [comment]
Hum I don't want to add `immutable` as a dependency, it's heavy and would affect everybody.

What you could do is check if `.get` and `.toJS` exist and are functions. We can assume that this will be a immutable object then

**Comment by evgeny-matveev** [comment]
@mathieudutour you're right, thanks. Is it ok now?

File: src/util.js [hunk 1]
```diff
@@ -11,6 +11,8 @@ export function getValuesForKey (key, item) {
           result.forEach(res => {
             tmp.push(res[_key])
           })
+        } else if (result instanceof Function) {
```
Comment (mathieudutour) [inline src/util.js]: isn't it `result.get` which should be a function?

Comment (evgeny-matveev) [inline src/util.js]: yes, my fault :(

**Comment by mathieudutour** [comment]
published in `react-search-input@0.9.0`. Thanks!

**Pull request merged**